python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -0,0 +1,225 @@
+          map.addIssue(CIDRIP, MESSAGE);
+          map.addIssue(CIDRIPV6, MESSAGE);
+        }
+      });
+  }
+
+  private static boolean isDictionaryWithAttributeBadProtocolEmptyIpAddressAdminPort(DictionaryAsMap map) {
+    return map.is(IPPROTOCOL, isString(SENSITIVE_PROTOCOL))
+      && (map.is(CIDRIP, isString(EMPTY_IPV4)) || map.is(CIDRIPV6, isString(EMPTY_IPV6)))
+      && map.inInterval("fromPort", "toPort", ADMIN_PORTS);
+  }
+
+  private static boolean isDictionaryWithAttributeInvalidProtocolEmptyIpAddress(DictionaryAsMap map) {
+    return map.is(IPPROTOCOL, isString(ANY_PROTOCOL))
+      && (map.is(CIDRIP, isString(EMPTY_IPV4)) || map.is(CIDRIPV6, isString(EMPTY_IPV6)));
+  }
+
+  // Class to handle Dictionary elements as a Map of ResolvedKeyValuePair, useful in case we need to refer to several elements and not only looking for a specific one.
+  // The keys are all resolved String in CdkUtils.ResolvedKeyValuePair.key
+  static class DictionaryAsMap implements CompareInterval {
+    Map<String, CdkUtils.ResolvedKeyValuePair> map = new HashMap<>();
+
+    public static DictionaryAsMap build(SubscriptionContext ctx, DictionaryLiteral dictionary) {
+      DictionaryAsMap dict = new DictionaryAsMap();
+
+      List<CdkUtils.ResolvedKeyValuePair> pairs = dictionary.elements().stream()
+        .map(e -> CdkUtils.getKeyValuePair(ctx, e))
+        .filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
+
+      for (CdkUtils.ResolvedKeyValuePair pair : pairs) {
+        pair.key.getExpression(isStringLiteral()).map(StringLiteral.class::cast)
+          .ifPresent(key -> dict.map.put(key.trimmedQuotesValue(), pair));
+      }
+      return dict;
+    }
+
+    public boolean is(String key, Predicate<Expression> valuePredicate) {
+      return map.containsKey(key) && map.get(key).value.hasExpression(valuePredicate);
+    }
+
+    public Optional<Expression> get(String key, Predicate<Expression> valuePredicate) {
+      if (!map.containsKey(key)) {
+        return Optional.empty();
+      }
+      return map.get(key).value.getExpression(valuePredicate);
+    }
+
+    public void addIssue(String key, String message) {
+      if (map.containsKey(key)) {
+        map.get(key).value.addIssue(message);
+      }
+    }
+
+    public Optional<Long> asLong(String name) {
+      return get(name, isNumericLiteral())
+        .map(NumericLiteral.class::cast)
+        .map(NumericLiteral::valueAsLong);
+    }
+  }
+
+  static class Call implements CompareInterval {
+    SubscriptionContext ctx;
+    CallExpression callExpression;
+
+    public Call(SubscriptionContext ctx, CallExpression callExpression) {
+      this.ctx = ctx;
+      this.callExpression = callExpression;
+    }
+
+    boolean hasArgument(String name, Predicate<Expression> predicate) {
+      return getArgument(ctx, callExpression, name).filter(flow -> flow.hasExpression(predicate)).isPresent();
+    }
+
+    public Optional<Long> asLong(String name) {
```

Comment:
This name also is not intuitive. How a call should be represented as long? Maybe `getArgumentAsLong(String argumentName)`